Makefile

```diff
@@ -154,7 +155,7 @@ kustomize: ## Download kustomize locally if necessary.
 	$(call go-get-tool,$(KUSTOMIZE),sigs.k8s.io/kustomize/kustomize/v3@v3.9.1)
 
 host-agent-binaries: ## Builds the binaries for the host-agent
-	RELEASE_BINARY=./byoh-hostagent GOOS=linux GOARCH=amd64 GOLDFLAGS="$(LDFLAGS) $(STATIC)" HOST_AGENT_DIR=./$(HOST_AGENT_DIR) $(MAKE) host-agent-binary
+	RELEASE_BINARY=./byoh-hostagent GOOS=linux GOARCH=amd64 GOLDFLAGS="$(LDFLAGS) $(STATIC) -X 'main.BuildVersion=${AGENT_BINARY_VERSION}' -X 'main.BuildTime=$(shell date)'" HOST_AGENT_DIR=./$(HOST_AGENT_DIR) $(MAKE) host-agent-binary
```

Comment:
Nitpick: Is it possible to convert the `shell date` format to `RFC3339`.

Reply:
Updated the output:

When the version is passed to Makefile:

```
bin/byoh-hostagent-linux-amd64 --version

agent version: version.Info{Major:"0", Minor:"1", Patch:"0", BuildDate:"2022-01-12T09:19:25Z", GoVersion:"go1.16.6", Platform:"linux/amd64", Compiler:"gc"}
```

When the version is not passed as default "dev" is used

```
bin/byoh-hostagent-linux-amd64 --version

agent version: version.Info{Major:"dev", Minor:"", Patch:"", BuildDate:"2022-01-12T09:19:25Z", GoVersion:"go1.16.6", Platform:"linux/amd64", Compiler:"gc"}
```